cineast-core/src/main/java/org/vitrivr/cineast/core/iiif/imageapi/ImageRequest.java

```diff
@@ -0,0 +1,147 @@
+package org.vitrivr.cineast.core.iiif.imageapi;
+
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.util.HashMap;
+import javax.imageio.ImageIO;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Class definition of a single IIIF Image API Request
+ *
+ * @author singaltanmay
+ * @version 1.0
+ * @created 29.05.21
+ */
+public class ImageRequest {
+
+  private static final Logger LOGGER = LogManager.getLogger();
+
+  private String baseUrl = null;
+  private String region = null;
+  private String size = null;
+  private String rotation = null;
+  private String quality = null;
+  private String extension = null;
+
+  public ImageRequest() {
+  }
+
+  public ImageRequest(String baseUrl, String region, String size, String rotation, String quality, String extension) {
+    this.baseUrl = baseUrl;
+    this.region = region;
+    this.size = size;
+    this.rotation = rotation;
+    this.quality = quality;
+    this.extension = extension;
+  }
+
+  /**
+   * Percent encodes "/","?","#","[","]","@" and "%" to their corresponding ASCII reserved characters
+   */
+  public static String percentEncode(String toEncode) {
```

Comment:
If I understand correctly, what you are attempting to do here is URL encoding. There are many more special characters that would need to be covered in this case. Luckily, there is already built-in logic for this, see java.net.URLEncoder.

Reply:
Replaced this method URLEncoder